inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp

```diff
@@ -0,0 +1,1350 @@
+                uni_vcvtps2dq(xmm_dst, xmm_dst);
+                uni_vpackusdw(xmm_dst, xmm_dst, xmm_dst);
+                uni_vpackuswb(xmm_dst, xmm_dst, xmm_dst);
+                pextrb(ptr[reg_dst], xmm_dst, 0);
+                break;
+            default:
+                assert(!"unknown dst_dt");
+        }
+    }
+
+    inline void horiz_ps(const Xmm& xmm, const Operand& op) {
+        switch (jcp_.reduce_mode) {
+            case Reduce::And:
+                andps(xmm, op);
+                break;
+            case Reduce::L1:
+            case Reduce::L2:
+            case Reduce::LogSum:
+            case Reduce::Mean:
+            case Reduce::Sum:
+            case Reduce::SumSquare:
+            case Reduce::LogSumExp:
+                addps(xmm, op);
+                break;
+            case Reduce::Max:
+                maxps(xmm, op);
+                break;
+            case Reduce::Min:
+                minps(xmm, op);
+                break;
+            case Reduce::Or:
+                orps(xmm, op);
+                break;
+            case Reduce::Prod:
+                mulps(xmm, op);
+                break;
+            default:
+                THROW_IE_EXCEPTION << "Unknown reduce mode.";
+        }
+    }
+};
+
+MKLDNNReduceNode::MKLDNNReduceNode(const InferenceEngine::CNNLayerPtr& layer, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache)
+        : MKLDNNNode(layer, eng, cache) {}
+
+void MKLDNNReduceNode::getSupportedDescriptors() {
+    if (!descs.empty())
+        return;
+
+    if (getParentEdges().size() != 2)
+        THROW_IE_EXCEPTION << "Incorrect number of input edges for layer " << getName();
+    if (getChildEdges().empty())
+        THROW_IE_EXCEPTION << "Incorrect number of output edges for layer " << getName();
+
+    if (getCnnLayer()->insData[REDUCE_DATA].lock()->getPrecision() != Precision::FP32 &&
+        getCnnLayer()->insData[REDUCE_DATA].lock()->getPrecision() != Precision::I32 &&
+        getCnnLayer()->insData[REDUCE_DATA].lock()->getPrecision() != Precision::U8) {
+        THROW_IE_EXCEPTION << "Incorrect input data tensor precision for layer " << getName() << ". Only FP32/I32/U8 are supported!";
+    }
+
+    if (getParentEdgeAt(REDUCE_INDEXES)->getDims().ndims() != 1) {
+        THROW_IE_EXCEPTION << "Index vector should be 1 dimension for layer " << getName();
+    }
+    if (getCnnLayer()->insData[REDUCE_INDEXES].lock()->getPrecision() != Precision::I32) {
+        THROW_IE_EXCEPTION << "Incorrect 'axes_to_reduction' input precision for layer " << getName() << ". Only I32 is supported!";
+    }
+
+    if (getCnnLayer()->outData[0]->getPrecision() != Precision::FP32 &&
+        getCnnLayer()->outData[0]->getPrecision() != Precision::I32 &&
+        getCnnLayer()->outData[0]->getPrecision() != Precision::U8) {
+        THROW_IE_EXCEPTION << "Incorrect output data tensor precision for layer " << getName() << ". Only FP32/I32/U8 are supported!";
```

Comment:
Is it possible to support I8 as well? I think it can be critical for quantized networks.

Reply:
Got it. I8 supported now. Thanks Dmitry!